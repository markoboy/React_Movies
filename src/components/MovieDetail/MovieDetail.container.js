import ErrorBoundary from '@components/ErrorBoundary';
import { MovieDetailType } from '@constants/MovieTypes';
import PropTypes from 'prop-types';
import React from 'react';
import MovieDetailComponent from './MovieDetail.component';

function MovieDetailContainer({ movie }) {
  return (
    <ErrorBoundary>
      <div className="background--dark-black margin--bottom">
        <MovieDetailComponent {...movie} />
      </div>
    </ErrorBoundary>
  );
}

MovieDetailContainer.propTypes = {
  movie: PropTypes.shape(MovieDetailType).isRequired,
};

export default MovieDetailContainer;
