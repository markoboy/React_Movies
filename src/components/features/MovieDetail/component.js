/* eslint-disable camelcase */
import ErrorBoundary from '@components/common/ErrorBoundary';
import Column from '@components/common/Grid/Column';
import Row from '@components/common/Grid/Row';
import Image from '@components/common/Image';
import withSection from '@components/hocs/WithSection';
import withWrapper from '@components/hocs/WithWrapper';
import { MOVIE_FALLBACK_IMAGE, MOVIE_POSTER_HEIGHT, MOVIE_POSTER_WIDTH } from '@constants/Generic';
import { MovieDetailType } from '@constants/MovieTypes';
import React, { memo } from 'react';
import './styles.scss';

const RowWithSectionWrapper = withSection(withWrapper(Row), {
  classes: 'section--large background--dark-black margin--bottom',
});

function MovieDetailComponent({
  runtime,
  vote_average,
  overview,
  poster_path,
  title,
  release_date,
  genres,
}) {
  return (
    <ErrorBoundary>
      <RowWithSectionWrapper>
        <Column classes="column--m-3">
          <Image
            src={poster_path}
            width={MOVIE_POSTER_WIDTH}
            height={MOVIE_POSTER_HEIGHT}
            alt={title}
            fallbackSrc={MOVIE_FALLBACK_IMAGE}
          />
        </Column>

        <Column classes="column--m-9">
          <div className="movie-detail">
            <div className="movie-detail__heading-container">
              <h1 className="movie-detail__heading">{title}</h1>
              {!!vote_average && (
                <p className="movie-detail__rating">{vote_average}</p>
              )}
            </div>

            <p className="movie-detail__genre">{genres.join(', ')}</p>

            <div className="movie-detail__year-container">
              {release_date && <p>{release_date.getFullYear()}</p>}
              {runtime && (
                <p className="movie-detail__duration">{`${runtime} min`}</p>
              )}
            </div>

            <p className="movie-detail__description">{overview}</p>
          </div>
        </Column>
      </RowWithSectionWrapper>
    </ErrorBoundary>
  );
}

MovieDetailComponent.propTypes = {
  ...MovieDetailType,
};

export default memo(MovieDetailComponent);
