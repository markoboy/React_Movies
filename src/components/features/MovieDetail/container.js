import { MOVIE_FALLBACK_IMAGE } from '@constants/Generic';
import { MovieDetailType } from '@constants/MovieTypes';
import useFallbackImage from '@hooks/UseFallbackImage';
import PropTypes from 'prop-types';
import React from 'react';
import MovieDetailComponent from './component';

function MovieDetailContainer({ movie }) {
  const { imageSrc, setFallbackImageSrc } = useFallbackImage(
    movie.poster_path,
    MOVIE_FALLBACK_IMAGE
  );

  return (
    <MovieDetailComponent
      {...movie}
      onError={setFallbackImageSrc}
      poster_path={imageSrc}
    />
  );
}

MovieDetailContainer.propTypes = {
  movie: PropTypes.shape(MovieDetailType).isRequired,
};

export default MovieDetailContainer;
