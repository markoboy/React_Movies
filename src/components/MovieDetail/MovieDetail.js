import Column from '@components/Grid/Column';
import Row from '@components/Grid/Row';
import { MOVIE_FALLBACK_IMAGE } from '@constants/Generic';
import { MovieDetailType } from '@constants/MovieTypes';
import useFallbackImage from '@hooks/UseFallbackImage';
import React, { memo } from 'react';
import './MovieDetail.scss';

function MovieDetail({
  runtime,
  vote_average: rating,
  overview,
  image,
  title,
  releaseDate,
  genres,
  fallbackImage,
}) {
  const { imageSrc, setFallbackImageSrc } = useFallbackImage(image, fallbackImage);

  return (
    <Row>
      <Column classes="column--m-3">
        <img src={imageSrc} alt={title} onError={setFallbackImageSrc} />
      </Column>

      <Column classes="column--m-9">
        <div className="movie-detail">

          <div className="movie-detail__heading-container">
            <h1 className="movie-detail__heading">{title}</h1>
            {!!rating && <p className="movie-detail__rating">{rating}</p>}
          </div>

          <p className="movie-detail__genre">{genres.join(', ')}</p>

          <div className="movie-detail__year-container">
            {releaseDate && <p>{releaseDate.getFullYear()}</p>}
            {runtime && <p className="movie-detail__duration">{`${runtime} min`}</p>}
          </div>

          <p className="movie-detail__description">{overview}</p>
        </div>
      </Column>
    </Row>
  );
}

MovieDetail.defaultProps = {
  fallbackImage: MOVIE_FALLBACK_IMAGE,
};

MovieDetail.propTypes = MovieDetailType;

export default memo(MovieDetail);
