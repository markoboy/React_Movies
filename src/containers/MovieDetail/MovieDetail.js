import React from 'react';
import PropTypes from 'prop-types';
// eslint-disable-next-line import/no-named-default
import { default as MovieDetailComponent } from '@components/MovieDetail';
import { MovieDetailType } from '@constants/MovieTypes';
import withWrapper from '@components/HigherOrder/WithWrapper';
import withSection from '@components/HigherOrder/WithSection';
import ErrorBoundary from '@containers/ErrorBoundary';

const MovieDetailComponentWithWrapper = withWrapper(
  withSection(MovieDetailComponent, { classes: 'section--large' })
);

function MovieDetail({ movie }) {
  return (
    <ErrorBoundary>
      <div className="background--dark-black margin--bottom">
        <MovieDetailComponentWithWrapper {...movie} />
      </div>
    </ErrorBoundary>
  );
}

MovieDetail.propTypes = {
  movie: PropTypes.shape(MovieDetailType).isRequired,
};

export default MovieDetail;
