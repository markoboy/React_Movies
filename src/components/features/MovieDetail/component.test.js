import React from 'react';
import { render } from '@testing-library/react';

import MovieDetail from './component';

describe('MovieDetail Component', () => {
  it('renders MovieDetail Component with provided props', () => {
    const onError = jest.fn();

    const { container } = render(
      <MovieDetail
        runtime={140}
        vote_average={8.7}
        overview="A great movie"
        poster_path="https://a-path.to/an/image.jpg"
        title="The best movie of all times"
        release_date={new Date()}
        genres={['Documentary']}
        onError={onError}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
