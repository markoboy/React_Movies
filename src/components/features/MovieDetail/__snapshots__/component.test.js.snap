// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail Component renders MovieDetail Component with provided props 1`] = `
<div>
  <section
    class="section section--large background--dark-black margin--bottom"
  >
    <div
      class="wrapper"
    >
      <div
        class="row"
      >
        <div
          class="column column--m-3"
        >
          <img
            alt="The best movie of all times"
            src="https://a-path.to/an/image.jpg"
          />
        </div>
        <div
          class="column column--m-9"
        >
          <div
            class="movie-detail"
          >
            <div
              class="movie-detail__heading-container"
            >
              <h1
                class="movie-detail__heading"
              >
                The best movie of all times
              </h1>
              <p
                class="movie-detail__rating"
              >
                8.7
              </p>
            </div>
            <p
              class="movie-detail__genre"
            >
              Documentary
            </p>
            <div
              class="movie-detail__year-container"
            >
              <p>
                2021
              </p>
              <p
                class="movie-detail__duration"
              >
                140 min
              </p>
            </div>
            <p
              class="movie-detail__description"
            >
              A great movie
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
